Rakefile

```diff
@@ -109,11 +109,28 @@ end
 
 desc 'Carthage tasks'
 namespace :carthage do
-  task :build_demo do
-    # Remove SPMTest app to prevent Carthage timeout
-    run! "rm -rf SampleApps/SPMTest"
-    run! "git add SampleApps"
+  def remove_spm_test_app
+    run! "mv SampleApps/SPMTest/ temp/"
+    run! "git add SampleApps/SPMTest"
     run! "git commit -m 'Remove SPMTest app to avoid Carthage timeout'"
+  end
+
+  # Remove SPMTest app to prevent Carthage timeout
+  task :remove_spm_test_app do
+    run! "mkdir temp"
+    remove_spm_test_app
+  end
+
+  # Restore SPMTest app to prevent Carthage timeout
+  task :restore_spm_test_app do
+    run! "mv temp/SPMTest/ SampleApps/"
+    run! "rm -rf temp/"
+    run! "git add SampleApps/SPMTest"
+    run! "git commit -m 'Restore SPMTest app for development'"
+  end
+
+  task :build_demo do
+    remove_spm_test_app
```

Comment:
random: Does `git reset --hard HEAD^` at the end of this task prevent us from having to call `restore_spm_test_app`?

Reply:
Exactly! For just building the demo_app, we don't want to commit the change and instead just RESET it.